vizro-core/tests/helpers/common.py

```diff
@@ -0,0 +1,48 @@
+import subprocess
+
+import cv2
+import imutils
+from hamcrest import assert_that, equal_to
+
+
+def comparison_logic(original_image, new_image):
+    """Comparison process."""
+    difference = cv2.subtract(original_image, new_image)
+    blue, green, red = cv2.split(difference)
+    assert_that(cv2.countNonZero(blue), equal_to(0), reason="Blue channel is different")
+    assert_that(cv2.countNonZero(green), equal_to(0), reason="Green channel is different")
+    assert_that(cv2.countNonZero(red), equal_to(0), reason="Red channel is different")
+
+
+def create_image_difference(original, new):
+    """Creates new image with diff of images comparison."""
+    diff = original.copy()
+    cv2.absdiff(original, new, diff)
+    gray = cv2.cvtColor(diff, cv2.COLOR_BGR2GRAY)
+    for i in range(0, 3):
+        dilated = cv2.dilate(gray.copy(), None, iterations=i + 1)
+    (t_var, thresh) = cv2.threshold(dilated, 3, 255, cv2.THRESH_BINARY)
+    cnts = cv2.findContours(thresh, cv2.RETR_LIST, cv2.CHAIN_APPROX_SIMPLE)
+    cnts = imutils.grab_contours(cnts)
+    for contour in cnts:
+        (x, y, width, height) = cv2.boundingRect(contour)
+        cv2.rectangle(new, (x, y), (x + width, y + height), (0, 255, 0), 2)
+    return new
+
+
+def compare_images(browserdriver, base_image, test_image_name):
```

Comment:
```suggestion
def assert_image_equal(browserdriver, base_image, test_image_name):
```

Please can we try to rewrite this function a bit to raise `AssertionError` rather than pure `Exception`? And to surround only minimal bits of code in `try/except`. Currently I'm not exactly what lines of code could raise which exception. e.g. something like this:
```
try:
    compare_images(original_new)
except AssertionError as exc:
    raise AssertionError("Images are different.") from exc
finally:
    # cleanup files
```

I'm not sure when `AttributeError` could be raised?

Reply:
I've rewrote the code to use more specific exceptions.
I can't use finally here, because every block of code under `try` and `except` is different and has its own logic.

Reply:
This looks much better now, thank you!